rust/src/types/amount.rs

```diff
@@ -370,10 +370,22 @@ impl Shr for u256 {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub struct U64Overflow;
+
 impl u256 {
     pub const ZERO: Self = Self(Uint256 { bytes: [0; 32] });
     pub const MAX: Self = Self(Uint256 { bytes: [0xff; 32] });
 
+    pub fn try_into_u64(self) -> Result<u64, U64Overflow> {
```

Comment:
This would be better modelled using [try_into](https://doc.rust-lang.org/beta/std/convert/trait.TryInto.html)
or try_from for uin64 (is this even possible?)

Is it there any literature against?